Tests/Controller/ContentTypeControllerTest.php

```diff
@@ -0,0 +1,114 @@
+    private $contentTypeGroupActionDispatcher;
+
+    /** @var \EzSystems\RepositoryForms\Form\ActionDispatcher\ActionDispatcherInterface|\PHPUnit_Framework_MockObject_MockObject */
+    private $contentTypeActionDispatcher;
+
+    /** @var \EzSystems\RepositoryForms\FieldType\FieldTypeFormMapperRegistryInterface|\PHPUnit_Framework_MockObject_MockObject */
+    private $fieldTypeMapperRegistry;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->contentTypeService = $this->getMock('eZ\Publish\API\Repository\ContentTypeService');
+        $this->searchService = $this->getMock('eZ\Publish\API\Repository\SearchService');
+        $this->userService = $this->getMock('eZ\Publish\API\Repository\UserService');
+        $this->contentTypeGroupActionDispatcher = $this->getMock('EzSystems\RepositoryForms\Form\ActionDispatcher\ActionDispatcherInterface');
+        $this->contentTypeActionDispatcher = $this->getMock('EzSystems\RepositoryForms\Form\ActionDispatcher\ActionDispatcherInterface');
+        $this->fieldTypeMapperRegistry = $this->getMock('EzSystems\RepositoryForms\FieldType\FieldTypeFormMapperRegistryInterface');
+    }
+
+    /**
+     * @return \EzSystems\PlatformUIBundle\Controller\ContentTypeController
+     */
+    protected function getContentTypeController()
+    {
+        $controller = new ContentTypeController(
+            $this->contentTypeService,
+            $this->searchService,
+            $this->userService,
+            $this->contentTypeGroupActionDispatcher,
+            $this->contentTypeActionDispatcher,
+            $this->fieldTypeMapperRegistry
+        );
+
+        return $controller;
+    }
+
+    public function getPrioritizedLanguageData()
+    {
+        return [
+            [
+                ['nor-NO'],
+                ['eng-GB', 'fre-FR'],
+                'eng-GB',
+                'eng-GB',
+            ],
+            [
+                ['nor-NO'],
+                ['eng-GB', 'fre-FR', 'nor-NO'],
+                'eng-GB',
+                'nor-NO',
+            ],
+            [
+                ['eng-GB', 'fre-FR', 'nor-NO'],
+                ['eng-GB', 'fre-FR'],
+                'eng-GB',
+                'eng-GB',
+            ],
+            [
+                ['nor-NO', 'eng-GB', 'fre-FR'],
+                ['eng-GB', 'fre-FR', 'nor-NO'],
+                'eng-GB',
+                'nor-NO',
+            ],
+            [
+                ['nor-NO', 'fre-FR', 'eng-GB'],
+                ['eng-GB', 'fre-FR'],
+                'eng-GB',
+                'fre-FR',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider getPrioritizedLanguageData
+     * @covers \EzSystems\PlatformUIBundle\Controller\ContentTypeController::getPrioritizedLanguage()
+     */
+    public function testGetPrioritizedLanguage($prioritizedLanguages, $languageCodes, $fallbackLanguageCode, $expected)
```

Comment:
Using reflection to unit test a protected / private method. We do not do that.

Being "forced" to do that is a clear case of an insufficient factoring. On the other hand, there was no unit testing whatsoever before, and you can't make it up. That's why I'm not asking that this is changed :-)